pdc_client/__init__.py

```diff
@@ -161,7 +258,13 @@ def decode(req):
 
         content_type = "application/json"
         encode = json.dumps
-        self.client = beanbag.BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        # In get_paged function it returns an interator, so user can get each pages quickly in plugins.
+        # The original call function of BeanBag returns the list.
+        # Keep the current codes and options in codes. Just init MultiPageBeanBag when is_multi is true.
+        if is_multi:
+            self.client = MultiPageBeanBag(url, session=self.session, fmt=(content_type, encode, decode))
```

Comment:
> If it's not possible to return different type (other than list) in the the new class, is it possible to handle this in in PDCClient.__getitem__()?
I think we can't choose the type in __getitem__, because in this function it returns the value of instance's key,  the type we are talking is about the return value of BeanBag.call, do you have any ideas about this part? 

> BTW, I cannot find BeanBag.make_request() in documentation and overriding the class methods is not mentioned at all. It could make upgrading to a new version difficult.

The make_request function is instance method, so you need init a instance first, and also this is used as a return of  call function in BeanBag.
`>>> import beanbag
>>> beanbag.__version__
'1.9.2'
>>> beanbag.BeanBag
<class 'beanbag.url_v1.BeanBag'>
>>> beanbag.BeanBag.make_reuqest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'BeanBag' has no attribute 'make_reuqest'
>>> a = beanbag.BeanBag
>>> a = beanbag.BeanBag()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: <lambda>() takes at least 2 arguments (1 given)
>>> a = beanbag.BeanBag("aaa","nn")
>>> a.make_request
<BeanBagBase(aaa/make_requestnn)>
>>> 
`